packages/story-editor/src/app/taxonomy/taxonomyProvider.js

```diff
@@ -191,11 +204,29 @@ function TaxonomyProvider(props) {
     (taxonomy, termSlugs = []) =>
       setSelectedSlugs((selected) => ({
         ...selected,
-        [taxonomy.rest_base]: termSlugs,
+        [taxonomy.rest_base]:
+          typeof termSlugs === 'function'
+            ? termSlugs(selected[taxonomy.rest_base])
+            : termSlugs,
       })),
     []
   );
 
+  // Fetch hierarchical taxonomies on mount
+  useEffect(() => {
+    // only fetch when `shouldRefetchCategories` is true
+    if (shouldRefetchCategories && taxonomies?.length) {
+      const hierarchicalTaxonomies = taxonomies.filter(
+        (taxonomy) => taxonomy.hierarchical
+      );
+      hierarchicalTaxonomies.forEach((taxonomy) =>
+        addSearchResultsToCache(taxonomy, { perPage: 100 })
+      );
+
+      setShouldRefetchCategories(false);
+    }
+  }, [addSearchResultsToCache, shouldRefetchCategories, taxonomies]);
```

Comment:
Reminder to self: talk to @littlemilkstudio on Monday about this.

The `addSearchResultsToCache` function updates the terms on the story directly using `updateStory`. This increments the version number in history.

Now the user will be prompted with the "your changes won't be saved" dialog if the user loads the editor but does nothing. Talk to max about it on monday to see what we could do here. History tests are failing 😭

Reply:
Added a ticket to fix this: https://github.com/google/web-stories-wp/issues/9073

Taxonomies will need to get history support.